dGame/dPropertyBehaviors/ControlBehaviors.h

```diff
@@ -19,15 +19,15 @@ class SystemAddress;
 typedef std::string BlockName;					//! A block name
 
 struct ControlBehaviorContext {
-	ControlBehaviorContext(AMFArrayValue* args, ModelComponent* modelComponent, Entity* modelOwner) noexcept
+	ControlBehaviorContext(const AMFArrayValue& args, ModelComponent* modelComponent, Entity* modelOwner) noexcept
 		: arguments{ args }, modelComponent{ modelComponent }, modelOwner{ modelOwner } {
 	};
 
 	operator bool() const {
-		return arguments != nullptr && modelComponent != nullptr && modelOwner != nullptr;
+		return modelComponent != nullptr && modelOwner != nullptr;
 	}
 
-	AMFArrayValue* arguments;
+	std::reference_wrapper<const AMFArrayValue> arguments;
```

Comment:
there a particular reason this cant be a reference?  the initializer list above should allow this correct?

Reply:
Using an actual const reference would make the struct non-moveable, only copyable, and a use of reference wrapper here doesn't require "get()," so it essentially has the semantic upside of representing non-nullability without any of the downsides of an actual reference

Reply:
I think its fine to have this be non move able.  This should be created once per request anyways

Reply:
Why change it? We're not even using .get() here. It's a perfect use case.

Reply:
I'll trust your judgment here in that case then